### DISCUSSION TITLE
Support key duplication in `SeedFactory.Test.produce/1`

### DISCUSSION THREAD

**Discussion by LostKobrakai** [fuelen/seed_factory#2] [Ideas]
It seems 
```
produce user: :user1, user: :user2
```
doesn't work, while doing the same in multiple lines works just fine.
```
produce user: :user1
produce user: :user2
```

I'd expect both to work just fine.

**Comment by fuelen** [comment]
I understand your expectation but I'm afraid it can't work due to more complex cases :slightly_smiling_face:
The order of entities specified to `produce` doesn't matter. You just declare what you need and SeedFactory automatically resolves which commands should be executed.
Let's take the schema from the README where we have company and user entities. If you type this
```elixir
produce [:user, company: :company1, company: :company2]
```
then which company the user should belong to? :slightly_smiling_face:

**Comment by LostKobrakai** [comment]
> then which company the user should belong to?

None of them would be my initial expectation. The company would be either taken from the `:company` (context) key, which none of the companies created in your produce call uses, or created implicitly. 

However you could always go with Keyword semantics and choose the first `:user` key in the keyword list given to produce. 

I can see not depending on the context map keys being useful to preventing a lot of „rekeying“ between entities.

**Comment by fuelen** [comment]
> None of them would be my initial expectation. The company would be either taken from the :company (context) key, which none of the companies created in your produce call uses, or created implicitly.

No, it doesn't work like this :slightly_smiling_face: 
```elixir
produce [:user, company: :company1]
```
is equivalent to
```elixir
setup ctx do
  rebind(ctx, [company: :company1], fn ctx ->
    # everything that is executed here will refer to :company1 instead of :company
    # if we produce :company entity, it will be produced as :company1
    # if :create_user command needs :company, it will be taken from the context using :company1 key
    ctx
    |> exec(:create_company)
    |> exec(:create_user)
  end)
end
```
